### PR TITLE
Extended annotations/endpoints - added request/response and some meta for OpenApi

### PR DESCRIPTION
**Pull request by benijo**

### CODE REVIEW

File: src/Annotation/Controller/Request.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	/** @var bool */
+	private $required = false;
+
+	/**
+	 * @param mixed[] $values
+	 */
+	public function __construct(array $values)
+	{
+		if (isset($values['description'])) {
+			$this->description = $values['description'];
```
Comment (mabar) [inline src/Annotation/Controller/Request.php]: `$this->description = $values['description'] ?? null`
Same in response please

File: src/Annotation/Controller/Responses.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+		}
+
+		$takenCodes = [];
+		/** @var Response $value */
+		foreach ($values['value'] as $value) {
+			if (!isset($takenCodes[$value->getCode()])) {
+				$takenCodes[$value->getCode()] = $value;
+			} else {
+				throw new AnnotationException(sprintf(
+					'Multiple @Response annotations with "code=%s" given. Each response must have unique code.',
```
Comment (mabar) [inline src/Annotation/Controller/Responses.php]: It should be primarily implemented as IValidation, annotations will be just one of the ways to define that. Feel free to just open an issue for me to do that :)

File: src/Annotation/Controller/Request.php (hunk 1 above)
Comment (benijo) [reply src/Annotation/Controller/Request.php]: done

File: src/Annotation/Controller/Responses.php (hunk 2 above)
Comment (benijo) [reply src/Annotation/Controller/Responses.php]: Mention here #72

**Pull request merged**